src/lib/mode_compiler.js

```diff
@@ -207,11 +207,25 @@ export function compileLanguage(language) {
     mode.__beforeBegin = null;
 
     mode.keywords = mode.keywords || mode.beginKeywords;
+
+    let kw_pattern = null;
+    if (typeof mode.keywords === "object") {
+      kw_pattern = mode.keywords.$pattern;
+      delete mode.keywords.$pattern;
```

Comment:
Can't figure out... why delete?

Reply:
So it doesn't get passed into the keyword compiler engine along with other groups.

Reply:
Ok, now I see.